eva/optimizer/optimizer_utils.py

```diff
@@ -239,19 +255,25 @@ def enable_cache(func_expr: FunctionExpression) -> FunctionExpression:
     Returns:
         FunctionExpression: The function expression with cache enabled.
     """
-    optimized_key = optimize_cache_key(func_expr)
-    if optimized_key == func_expr.children:
-        optimized_key = [None]
+    cache = enable_cache_init(func_expr)
+    return func_expr.copy().enable_cache(cache)
 
-    name = func_expr.signature()
-    cache_entry = CatalogManager().get_udf_cache_catalog_entry_by_name(name)
-    if not cache_entry:
-        cache_entry = CatalogManager().insert_udf_cache_catalog_entry(func_expr)
 
-    cache = FunctionExpressionCache(
-        key=tuple(optimized_key), store=DiskKVCache(cache_entry.cache_path)
+def enable_cache_on_tree(expr_tree: AbstractExpression):
+    func_exprs = list(expr_tree.find_all(FunctionExpression))
+    func_exprs = list(filter(lambda expr: check_expr_validity(expr), func_exprs))
+    for expr in func_exprs:
+        cache = enable_cache_init(expr)
```

Comment:
Please change the function name. How about `enable_cache_on_function_expression`?

Reply:
will change to `enable_cache_on_function_expression`